### PR TITLE
API to get an existing AI analysis

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Adds an endpoint to fetch an AI analysis for a given student and learning goal. 

Specific things I'd love feedback on:
- This API is currently per learning goal. Does that feel right or should it be by rubric?
- Right now the row of the database is returned as-is. I couldn't really think of what else we'd need but let me know if I'm missing something!

### CODE REVIEW

**Comment by davidsbailey** [comment]
Very exciting! I'll have more to offer once my half of this is in place (writing to this table), but for now I would say:

> * This API is currently per learning goal. Does that feel right or should it be by rubric?

I think we'll always want to read the entire rubric, so having the api be by rubric makes more sense to me.

> * Right now the row of the database is returned as-is. I couldn't really think of what else we'd need but let me know if I'm missing something!

I think what we need is, for each learning goal, what is the `understanding` from the corresponding ai evaluation. I would suggest returning only what we need, but it's possible my thoughts on server APIs have become out of date with our best practices. Then we can add any other fields to a `summarize` method as we need them?

**Comment by bethanyaconnor** [comment]
> I think what we need is, for each learning goal, what is the understanding from the corresponding ai evaluation. I would suggest returning only what we need

Done in 1c92c16839c! I kept the id in there -- I find it can be useful to have a unique identifier available on the client -- but can remove if it feels too extraneous.

> I think we'll always want to read the entire rubric, so having the api be by rubric makes more sense to me.

Where I was struggling with this is that the teacher evaluations are fetched by learning goal (correct me if I'm wrong @Nokondi) and want to keep them the same. On the other hand, the teacher evaluations are dynamic (at least in the teacher view), versus the AI evaluations where, except in the case of rerunning the evaluation, are static within the session.

Alright, I think I convinced myself to move this to rubrics controller -- I'll work on that now. @kobryan0619 I'm interested if you have any thoughts here as you've currently got the task to use this API.

**Comment by davidsbailey** [comment]
> > I think we'll always want to read the entire rubric, so having the api be by rubric makes more sense to me.
> 
> Where I was struggling with this is that the teacher evaluations are fetched by learning goal (correct me if I'm wrong @Nokondi) and want to keep them the same. On the other hand, the teacher evaluations are dynamic (at least in the teacher view), versus the AI evaluations where, except in the case of rerunning the evaluation, are static within the session.
> 
> Alright, I think I convinced myself to move this to rubrics controller -- I'll work on that now. @kobryan0619 I'm interested if you have any thoughts here as you've currently got the task to use this API.

haha yes, I am running into the same thing on the other side in response to the openai request. what I want is to write all the LearningGoalAiEvaluation objects at the same time, but there is no parent object to nest them under so I am just writing them individually. in your case I think it will be easier to bulk read them than it is for me to bulk write them, so you get to decide :-) if it is harder / not convenient to bulk read then I think what you have here is also fine.

**Comment by Nokondi** [comment]
@bethanyaconnor, you're right that the LGEs are retrieved by learning goal id. Since everything is tied to the rubric, would it help to fetch learning goals and evals within the rubric controller? I don't even know if that makes sense for what you're doing, but it would mean fewer backend queries, since the rubric controller could pull together all of the parts of the rubric, send them to the front end, then the appropriate parts can be sent to child components as props.

**Comment by bethanyaconnor** [comment]
Thanks all! I think bulk reading the AI evaluations makes sense and will require fewer network requests from the client as well. I already made that change but looks like I forgot to push it 🤦‍♀️

The teacher side is a bit trickier to me. It might make sense for us to bulk fetch them on page load, but then updates make sense to be per-learning goal. I think, as a future action item, we can think about cleaning that up but that's definitely out of scope of this PR and the follow up integration task.

File: dashboard/app/models/learning_goal_ai_evaluation.rb
Comment (davidsbailey) [inline dashboard/app/models/learning_goal_ai_evaluation.rb:1]: 👋

File: dashboard/app/controllers/rubrics_controller.rb [hunk 1]
```diff
@@ -55,6 +55,20 @@ def submit_evaluations
...
+  def get_ai_evaluations
+    return head :forbidden unless current_user&.teacher?
+
+    permitted_params = params.transform_keys(&:underscore).permit(:id, :student_id)
+    student = User.find(permitted_params[:student_id])
+    return head :not_found unless student
+    return head :forbidden unless can?(:manage, student)
+
+    learning_goals = LearningGoal.where(rubric_id: permitted_params[:id])
+    learning_goal_ai_evaluations = LearningGoalAiEvaluation.where(user_id: permitted_params[:student_id], learning_goal_id: learning_goals.pluck(:id))
```
Comment (davidsbailey) [inline dashboard/app/controllers/rubrics_controller.rb]: would we also want to filter to just the most recent evaluation for each learning goal? this is what I imagine we'll need to populate the new rubric UI, but I could be missing how this is going to be used exactly

File: dashboard/test/controllers/rubrics_controller_test.rb [hunk 2]
```diff
@@ -66,4 +66,43 @@ class RubricsControllerTest < ActionController::TestCase
...
+    student = create :student
+    teacher = create :teacher
+    create :follower, student_user: student, user: teacher
+    sign_in teacher
+
+    rubric = create :rubric
+    learning_goal1 = create :learning_goal, rubric: rubric
+    learning_goal2 = create :learning_goal, rubric: rubric
+    ai_evaluation1 = create :learning_goal_ai_evaluation, learning_goal: learning_goal1, user: student, understanding: 1
+    ai_evaluation2 = create :learning_goal_ai_evaluation, learning_goal: learning_goal2, user: student, understanding: 2
```
Comment (davidsbailey) [inline dashboard/test/controllers/rubrics_controller_test.rb:80]: not blocking for this PR, but we should probably add constants to refer to these understanding values. looks like we actually have some already, thoughts on whether to reuse these? https://github.com/code-dot-org/code-dot-org/blob/472deb8c693c54512379775c6c9a4e09a99952dc/lib/cdo/shared_constants.rb#L601-L608

File: dashboard/test/controllers/learning_goal_ai_evaluations_controller_test.rb
Comment (davidsbailey) [inline dashboard/test/controllers/learning_goal_ai_evaluations_controller_test.rb:1]: remove this file too?

**Review by davidsbailey — APPROVED** [review]

File: dashboard/test/controllers/rubrics_controller_test.rb (hunk 2 above)
Comment (bethanyaconnor) [reply dashboard/test/controllers/rubrics_controller_test.rb:80]: oh yep! Was planning on using those but was planning on doing so in a follow-up PR if that's alright?

File: dashboard/app/controllers/rubrics_controller.rb (hunk 1 above)
Comment (bethanyaconnor) [reply dashboard/app/controllers/rubrics_controller.rb]: oh good call!

File: dashboard/app/models/learning_goal_ai_evaluation.rb
Comment (bethanyaconnor) [reply dashboard/app/models/learning_goal_ai_evaluation.rb:1]: ooo I shouldn't have deleted the model -- only the controller. I'll add this back

**Pull request merged**